src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/PseudoCustomAttributesData.cs

```diff
@@ -186,9 +183,9 @@ internal BlobBuilder SerializeMarshallingData()
                     {
                         writer.WriteCompressedInteger((int)safeArrayElementSubtype);
 
-                        if (_marshalTypeNameOrSymbol is Type elementType)
+                        if (_marshalTypeName is string)
                         {
-                            writer.WriteSerializedString(elementType.FullName);
+                            writer.WriteSerializedString(_marshalTypeName);
                         }
```

Comment:
Does this need to have an else clause to write 0 when the string is null? (Pre-existing bug.)

Reply:
Based on the other nullable ref type handling looks you are right